sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/SearchIndexClientBuilder.java

```diff
@@ -36,16 +36,68 @@
 
 /**
  * This class provides a fluent builder API to help aid the configuration and instantiation of {@link SearchIndexClient
- * SearchIndexClients} and {@link SearchIndexAsyncClient SearchIndexAsyncClients}. Call {@link #buildClient()
- * buildClient} and {@link #buildAsyncClient() buildAsyncClient} respectively to construct an instance of the desired
- * client.
+ * SearchIndexClients} and {@link SearchIndexAsyncClient SearchIndexAsyncClients}.
+ *
+ * <h2>
+ *     Overview
+ * </h2>
+ *
+ * <p>
+ *     This client allows you to create instances of {@link SearchIndexClient} and {@link SearchIndexAsyncClient} to
+ *     utilize synchronous and asynchronous APIs respectively to interact with Azure AI Search.
+ * </p>
+ *
+ * <h2>
+ *     Getting Started
+ * </h2>
+ *
+ * <h3>
+ *     Authentication
+ * </h3>
+ *
+ * <p>
+ *     Azure AI Search supports <a href="https://learn.microsoft.com/en-us/azure/search/search-security-rbac?tabs=config-svc-portal%2Croles-portal%2Ctest-portal%2Ccustom-role-portal%2Cdisable-keys-portal">
+ *         Microsoft Entra ID (role-based) authentication </a> and <a href="https://learn.microsoft.com/en-us/azure/search/search-security-api-keys?tabs=portal-use%2Cportal-find%2Cportal-query">API keys</a> for authentication.
+ * </p>
+ *
+ * <p>
+ *     For more information about the scopes of authorization, see the <a href="https://learn.microsoft.com/azure/search/search-security-overview#authentication">Azure AI Search Security Overview</a> documentation.
+ * </p>
+ *
+ * <h4>
+ *     Building and Authenticating a {@link SearchIndexClient} or {@link SearchIndexAsyncClient} using API keys
+ * </h4>
+ *
  * <p>
- * The following must be provided to construct a client instance.
+ *     To build an instance of {@link SearchIndexClient} or {@link SearchIndexAsyncClient} using API keys, call
```

Comment:
Should add a similar mention here as done in the Microsoft Entra credentials as Search has reader API keys and admin API keys which can limit the access permitted.